Will no longer build on Solaris unless __EXTENSIONS__ is also defined

go-sqlite3 no longer builds on Solaris. Adding CGO_CFLAGS -D__EXTENSIONS__=1 fixes the issue.

    mordac ~/go>go get -f -t -u -v github.com/mattn/go-sqlite3
    github.com/mattn/go-sqlite3 (download)
    github.com/mattn/go-sqlite3
    # github.com/mattn/go-sqlite3
    sqlite3-binding.c:30887:43: error: ‘ioctl’ undeclared here (not in a function)
       { "ioctl",         (sqlite3_syscall_ptr)ioctl,          0 },
                                               ^~~~~
    mordac ~/go>env CGO_CFLAGS=-D__EXTENSIONS__=1 go get -f -t -u -v github.com/mattn/go-sqlite3
    github.com/mattn/go-sqlite3 (download)
    github.com/mattn/go-sqlite3
    mordac ~/go>git show-ref HEAD
    mordac ~/go>(cd src/github.com/mattn/go-sqlite3/; git show-ref HEAD)
    6c771bb9887719704b210e87e934f08be014bdb1 refs/remotes/origin/HEAD
    mordac ~/go>uname -a
    SunOS mordac 5.11 joyent_20170302T081240Z i86pc i386 i86pc

The root cause is that on Solaris `ioctl()` is defined in stropts.h for XPG compliance, and the unistd.h version is not defined if the XPG macros are defined unless __EXTENSIONS__ is also defined.

    mordac ~/go>ggrep -B 4 -A 3 ioctl /usr/include/unistd.h
    #if !defined(__XOPEN_OR_POSIX) || defined(__EXTENSIONS__)
    extern char *getusershell(void);
    #endif /* !defined(__XOPEN_OR_POSIX) || defined(__EXTENSIONS__) */
    /*
     * The following ioctl prototype is duplicated in <stropts.h>. The
     * duplication is necessitated by XPG4.2 which requires the prototype
     * be defined in <stropts.h>.
     */
    #if !defined(__XOPEN_OR_POSIX) || defined(__EXTENSIONS__)
    extern int ioctl(int, int, ...);
    extern int isaexec(const char *, char *const *, char *const *);
    extern int issetugid(void);
    #endif

--- a/sqlite3_solaris.go
+++ b/sqlite3_solaris.go
@@ -1,0 +1,12 @@
+// Copyright (C) 2018 Yasuhiro Matsumoto <mattn.jp@gmail.com>.
+//
+// Use of this source code is governed by an MIT-style
+// license that can be found in the LICENSE file.
+// +build solaris
+
+package sqlite3
+
+/*
+#cgo CFLAGS: -D__EXTENSIONS__=1
+*/
+import "C"

